src/metal.cs

```diff
@@ -393,10 +404,24 @@ partial interface MTLCommandBuffer {
 		void PopDebugGroup ();
 
 #if NET
+#if !TVOS
+		[MacCatalyst (14, 0), Mac (11, 0), iOS (13, 0), NoTV]
 		[Abstract] // @required but we can't add abstract members in C# and keep binary compatibility
+#else
+#if XAMCORE_5_0
+		[Abstract]
+#endif
+		[NoMacCatalyst, NoMac, NoiOS, TV (16,0)]
+
+#if XAMCORE_5_0
+		[Abstract]
+#endif
+
+#endif
+
+#else
+		[MacCatalyst (14, 0), Mac (11, 0), iOS (13, 0), TV (16, 0)]
```

Comment:
This is not right, we won't end up with the correct availability attributes in XAMCORE_5_0: the Microsoft.tvOS.dll version will say it's the only platform, while the other platforms will say they all exist except tvOS. This is what we'd end up with once we define XAMCORE_5_0:

```cs
#if !TVOS
		[Abstract]
		[MacCatalyst (14, 0), Mac (11, 0), iOS (13, 0), NoTV]
#else
		[Abstract]
		[Abstract]
		[NoMacCatalyst, NoMac, NoiOS, TV (16,0)]
#endif
```

(there will be two [Abstract] attributes for !TVOS as well)

This is simpler and correct I think:

```cs
#if XAMCORE_5_0
		[Abstract]
		[MacCatalyst (14, 0), Mac (11, 0), iOS (13, 0), TV (16, 0)]
#elif !TVOS
		[Abstract]
		[MacCatalyst (14, 0), Mac (11, 0), iOS (13, 0), NoTV]
#else
		[NoMacCatalyst, NoMac, NoiOS, TV (16,0)]
#endif
```